src/stcal/alignment/util.py

```diff
@@ -115,12 +131,14 @@ def _generate_tranform(
     transform : ~astropy.modeling.Model
         An :py:mod:`~astropy` model containing the transform between frames.
     """
+    if not isinstance(wcsinfo, dict):
+        wcsinfo = wcsinfo.instance
```

Comment:
I just tested a roman `cal` file from artifactory that has a wcsinfo.

The wcsinfo object fails an `isinstance(wcsinfo, dict)` check (it is of type `roman_datamodels.stnode.Wcsinfo`).

If I switch this to `isinstance(wcsinfo, WcsInfo)` it fails for 2 reasons:
- missing `instance` method
- missing `wcsaxes`

Reply:
Good to know.  What do you think is the best way to handle this?  I agree maybe checking `isinstance(wcsinfo, WcsInfo)` is a good plan, then checking `isinstance(wcsinfo, dict)` and if it's neither, then perhaps raise an error.  But that doesn't solve the issue that jwst and roman have different ways of accessing `wcsinfo` as a dict. (as an aside, I find this a bit strange).  I could check for either `.instance` or `.to_flat_dict`, but I'm not sure how to set up a protocol to do that, or if that's even a good idea.

Regarding `wcsaxes`, that attribute is used only in the function `wcsinfo_from_model`.  Looking again, I am not sure why `wcsinfo_from_model()` ended up in stcal in the first place... I don't see it imported in romancal nor jwst, and jwst has its own version of it inside `pointing.py` that looks identical (except the changes I made that remove dependence on the datamodel structure).  This was introduced in [this commit](https://github.com/spacetelescope/stcal/commit/0a1fa458d1a9dd6b6ba5dcd392e258ec79509383) related to resample. @mairanteodoro can you confirm whether or not this is still needed?  If not, I will go ahead and remove the function entirely.

Reply:
At the moment I'm not sure what to recommend and would have to dig into the code to come up with one.

Ideally this PR wouldn't break romancal. Is that possible?

Reply:
@braingram The root cause of all the ways in which this will break romancal (and jwst for that matter) is that the call structure of certain `stcal.alignment.util` functions was changed to remove the need for the `SupportsDataWithWCS` protocol.  Some of these functions are used in resample as well as tweakreg, namely `wcs_from_footprints()` and `update_s_region_imaging()`.

My takeaway from Thursday's meeting was that people were satisfied with those changes, as that protocol was not very well-defined and we would like to remove dependency on the datamodel structure.  It sounded from yesterday's meeting like there was no problem with making a patch of romancal that pinned an upper dependency of stcal and then modifying those function calls as part of @mairanteodoro's work to make the romancal tweakreg step rely on stcal.  Is it no longer the case that this solution is okay?

Of course other ways of doing this are are possible: for example, to @mcara's point, I could write basically a datamodel-to-wcs-and-wcsinfo interface, and then revert all the call structure changes.  I view that solution as a clunky and non-permanent fix, and it seems to me it would have to be changed again as soon as we start moving some of these functions to tweakwcs and gwcs.

Making it so the Wcsinfo object I introduced into stcal can use either the `to_flat_dict()` or `.instance()` method to return a dictionary I think is a good idea, and I'll go ahead with that.  However, I still think it's strange that they are different between jwst and romancal, and I wonder if we should consider changing one of them to make them consistent.

Reply:
FWIW, we did decide yesterday at the romancal tag to do a new romancal release with an upper pin on stcal to enable this.

Reply:
> My takeaway from Thursday's meeting was that people were satisfied with those changes, as that protocol was not very well-defined and we would like to remove dependency on the datamodel structure. In the meeting it sounded like there was no problem with making a patch of romancal that pinned an upper dependency of stcal and then modifying those function calls as part of @mairanteodoro's work to make the romancal tweakreg step rely on stcal. Is it no longer the case that this solution is okay?

It's my understanding that solution is still ok. However the use of `wcsinfo.instance` meant that to fix romancal would involve either:
- removing the `wcsinfo.instance` use here in stcal (which is being introduced with this PR)
- fundamental changes to roman datamodels (which would be no small undertaking)

I see that you've updated the PR to remove the `instance` use. Thanks for updating the PR. I only brought it up because I tested this with a roman file and encountered an issue that would come up when attempting to update romancal.

Reply:
Ok great, glad we are on the same page.  I got confused and assumed you were still concerned about `wcs_from_footprints`.

There should hopefully be no need to change roman datamodels now, and I think the `Wcsinfo` protocol is now as simple as possible while still being specific about its needs.